src/client/mod.rs

```diff
@@ -70,14 +88,13 @@ impl ResponseStream {
     }
 }
 
-impl StateStream for ResponseStream {
+impl<E> StateStream for ResponseStream<E> where E: ImapClient {
     type Item = ResponseData;
-    type State = Client;
+    type State = E::Original;
```

Comment:
If you got rid of `Original`, this would just be `E`.

Reply:
Yup, this is gone now.